### ISSUE TITLE
F6 navbar outline and tooltip

### ISSUE THREAD

**Issue by olefredrik** [zurb/foundation-sites#7028]
Hello everyone in Zurb! First, congratulations on the successful release of Foundation 6. Massive respect for all the hard work you have put into this!

I'm experiencing weird behavior on the navbar elements displaying tootips on menu-item hover and outlines on dropdowns. See the attached screendump. Mac OSX 10.11.1, Chrome 46.0.2490.86.

![tooltip-and-outline](https://cloud.githubusercontent.com/assets/1080017/11284831/35779c8e-8f0c-11e5-8550-5e7c08ffce3c.png)

The problem seems to disappear if I remove tabindex attributes on the navbar elements. I am aware that tabindex makes it easier to navigate with keyboard and has certain advantages in terms of accessibility. But I'm questioning if it's worth it, if it causes visual noise when using a mouse or trackpad (which I assume most desktop users do?) 

It would be quite cool if we could keep tabindex and display an outline on the focused only when pressing the tab key on the keyboard.

**Comment by zurbchris** [comment]
We're using a package: what-input, to curtail that. Though admittedly it's not perfect, the outlines are a part of our A11y support.

**Comment by moondawg69** [comment]
Yes, seeing that as well. But the Outline doesn't 'work' on Safari 9.0.2 for instance, it breaks as you move and is very disconcerting. Also the hover doesn't correspond with the item you are hovering over. My cursor was over 2 and 3, but it displays 1. Seen on http://foundation.zurb.com/sites-templates-previews/product-page.html

<img width="320" alt="screen shot 2015-11-20 at 00 20 16" src="https://cloud.githubusercontent.com/assets/1304000/11288507/8ab26872-8f1c-11e5-9cc9-a28bf85d14c4.png">

**Comment by erredeco** [comment]
this works?
```
element:focus{outline:none}
```

**Comment by andycochran** [comment]
Please don't `outline:none`. :cry: 
The [what-input](http://ten1seven.github.io/what-input/) utility is included as to not cripple accessibility.

**Comment by imranomari** [comment]
there two issues here:
1 - the outline, is there a standard way to hide it without affecting a11y?
2 - the annoying tooltip, which appears with wrong content usually, as mentioned above it contains the text of the first list item, even when hovering the second or the third item

**Comment by zurbchris** [comment]
Check the body tag in your browser dev console and see if `[data-whatinput]` is updating when you use the mouse or switch to the keyboard. There will also be a patch to fix the hover selection issue, I'm working on it now.
As for the tooltip - that is browser specific and rendered based on the `[title="example"]` attribute. If you don't want it, don't use a title. They aren't greatly supported by assistive technologies anyway.

**Comment by imranomari** [comment]
@zurbchris I don't think that the tooltip is rendered based on the title attribute, I tried it on dropdown menu without the title attribute in any of the anchors, and it still shows the tooltip, empty.

**Comment by olefredrik** [comment]
@zurbchris : Aha. Smart solution. The ```[data-whatinput]``` attribute changes when I switch between using mouse and keyboard.

@imranomari : If I remove the title attribute from the menu-item list element, the tooltip is gone (at least on my mac / latest version of chrome)

**Comment by zurbchris** [comment]
@imranomari you can't just empty the attribute, you have to remove it completely from your markup, use the `aria-label="Example Content"` attribute instead.

**Comment by imranomari** [comment]
@zurbchris yes ofc, no title attribute at all, tested with this code below:
```html
<html class="no-js" lang="ar" dir="rtl">

<!-- .. -->

<div class="top-bar">
    <div class="row">
        <div class="top-bar-right">
            <ul class="dropdown menu align-right" data-dropdown-menu data-disable-hover="true">
                <li class="menu-text">Yeti Store</li>
                <li class="has-submenu">
                    <a href="#">One</a>
                    <ul class="submenu menu vertical" data-submenu>
                        <li><a href="#">One</a></li>
                        <li><a href="#">Two</a></li>
                        <li><a href="#">Three</a></li>
                    </ul>
                </li>
                <li><a href="#">Two</a></li>
                <li><a href="#">Three</a></li>
            </ul>
        </div>
        <div class="top-bar-left">
              <!-- ... -->
        </div>
    </div>
</div>
```



@olefredrik here is a gif for a demo  in ios 10.11.1, Chrome 46.0.2490.86
![tooltip_demo](https://cloud.githubusercontent.com/assets/2339897/11312041/7d5e2076-8fcb-11e5-9f37-7871252a7db8.gif)

**Comment by imranomari** [comment]
@zurbchris check this line
https://github.com/zurb/foundation-sites/blob/v6.0.1/js/foundation.dropdownMenu.js#L145
```js
    this.$tabs.each(function(){
      var $tab = $(this);
      $tab.attr({
        'role': 'menuitem',
        'tabindex': 0,
        'title': $tab.children('a:first-child').text()/* <== this guy here */
      }).children('a').attr('tabindex', -1);//maybe ....
      if($tab.children('[data-submenu]')){
        $tab.attr('aria-haspopup', true);
      }
    });
```

**Comment by zurbchris** [comment]
see https://github.com/zurb/foundation-sites/issues/7063#issuecomment-158511519

**Comment by imranomari** [comment]
Great, thank you for your efforts.

**Comment by imranomari** [comment]
@zurbchris back to the outline issue, it seems that toggling the dropdown menu using the mouse doesn't set that data-whatinput attribute, which makes the outline always visible when the dropdown is active even with `disable-mouse-outline` mixin included in `.has-menu`.

when I set the data-whatinput value directly from the console, only then the outline is set to 0;

**Comment by zurbrandon** [comment]
This has been all fixed up in the 6.0.2 patch.
